docs/content/guides/dagster/ml-ops.mdx

```diff
@@ -0,0 +1,221 @@
+---
+title: Building your machine learning pipelines In Dagster | Dagster Docs
+description: This guide illustrates how to use Dagster to operationaize your machine learning pipeline
+---
+
+# Building your machine learning pipelines with Dagster
+
+In this guide, we’ll walk you through how to take your machine learning models and deploy and mantain them in production using Dagster, reliably and efficently.
+
+<Note>
+  This guide assumes you have familiarity with machine learning concepts,{" "}
+  <a href="/tutorial">the Dagster tutorial</a> and several Dagster concepts,
+  including{" "}
+  <a href="/concepts/assets/software-defined-assets">software-defined assets</a>{" "}
+  and <a href="/concepts/ops-jobs-graphs/jobs">jobs</a> . We also recommend
+  reviewing the{" "}
+  <a href="/guides/dagster/automating-pipelines">
+    automating data pipelines guide
+  </a>{" "}
+  as many of the same principles can be applied to MLOps.
+</Note>
+
+We will work through several aspects of MLOps, including using assets for different elements of your machine learning pipeline, how to automate model training and monitoring your model's drift.
+
+---
+
+## Why build your ML pipelines with Dagster?
+
+- Dagster is designed to use during development, the lightweight execution model means you can access the benefits of an orchestrator, like re-executing from the middle of a pipeline and parallelizing steps while you're experiementing.
+- It models data assets, not just tasks so Dagster understands the upstream and downstream data depenedencies and is a one-stop shop for both the data transformations and the models that depend on the data transformations.
+
+---
+
+## Machine learning development
+
+Before we go into how to operatialize a machine learning model, let's go over some of the tools you can use in Dagster to develop a machine learning model.
+
+If you are already using Dagster for your ETL pipelines, it is a natural progression to build out and test your models in Dagster.
+
+For this guide, we will be using the hackernews data which is demoed in the [tutorial](/tutorial).
+
+The machine learning model we will walk through is taking the hackernews stories, using the title to predict the number of comments that a story will generate. This will be a supervised model since we have the number of comments for all the previous stories.
+
+The assets graph will look like this at the end of this guide:
+
+<Image
+alt="alt"
+src="/images/guides/ml-ops/ml_asset_dag.png"
+width={1804}
+height={1064}
+/>
+
+### Data Ingestion
+
+First, we will be creating an asset which retrieves all the hackernews records from the previous run to the latest hackernews record.
```

Comment:
This isn't exactly accurate, correct? In the sense that we're not retrieving all the records from the previous run?

Reply:
updated